core/src/main/java/hudson/model/Queue.java

```diff
@@ -866,7 +841,7 @@ private WaitingItem peek() {
      * @since 1.483
      */
     public List<Item> getApproximateItemsQuickly() {
-        return itemsView.get();
+        return getItemsAsList();
```

Comment:
What is wrong with `Arrays.asList(getItems())` (which avoids the use of the builder which will generate the same / more temporary objects on the heap) ?